### PR TITLE
Changes HTTP response from 500 to 502 BAD_GATEWAY on ES down

### PR DESCRIPTION
**Pull request by bdc34**
Change to differentiate between ES down and other errors.

ARXIVNG-5112

### CODE REVIEW

**Review by bmaltzan — APPROVED** [review]
Looks good.

**Comment by bdc34** [comment]
I've also added the library updates for pyjwt and lxml suggested by dependbot.

File: search/controllers/advanced/__init__.py [hunk 1]
```diff
@@ -143,8 +139,13 @@ def search(request_params: MultiDict) -> Response:
                 ) from ex
             except index.OutsideAllowedRange as ex:
                 raise BadRequest(
-                    "Hello clever friend. You can't get results in that range"
-                    " right now."
+                    "You can't get results in that range."
+                ) from ex
+            except Exception as ex:
+                logger.error("Unhandled exception: %s", str(ex))
+                raise InternalServerError(
```
Comment (DavidLFielding) [inline search/controllers/advanced/__init__.py:146]: Do we want to return an internal server error here?

**Review by DavidLFielding — COMMENTED** [review]
Looks good. My only question is whether the ES application should ever return internal server error.

File: search/controllers/advanced/__init__.py (hunk 1 above)
Comment (bdc34) [reply search/controllers/advanced/__init__.py:146]: Yes this should return a 500. This is the case where there is an exception that was unexpected and we want to announce that as a 500 so we can notice it and fix it.

**Pull request merged**